src/app/upload/upload.component.html

```diff
@@ -32,13 +32,38 @@ <h3 class="Color--primary Line--Break">{{additionalMessage}}</h3>
     </div>
   </div>
   <br/>
-  <div *ngIf="kitType != null" class="Width--80">
+  <div *ngIf="uploadReasons.length > 0">
+    <div>
+      <b>Select reason to upload this batch</b>
+    </div>
+    <div>
+      <md-select [(ngModel)]="this.selectedReason">
+        <md-option *ngFor="let uploadReason of this.uploadReasons" value="{{uploadReason}}">{{uploadReason}}</md-option>
+      </md-select>
+    </div>
+  </div>
+  <br>
+  <div *ngIf="carriers.length > 0">
+    <div>
+      <b>Select Shipping Carrier</b>
+    </div>
+    <div>
+      <md-select [(ngModel)]="this.selectedCarrier">
+        <md-option *ngFor="let carrier of this.carriers" value="{{carrier}}">{{carrier}}</md-option>
+      </md-select>
+    </div>
+  </div>
+  <br/>
+  <div *ngIf="kitType != null && (uploadReasons.length == 0 || selectedReason !== null)  && (carriers.length == 0 || selectedCarrier !== null)" class="Width--80">
```

Comment:
Maybe move this expression into a validation method in the component directly?  Then other places can make the same assertion (and the ts can be tested in isolation).